Fixes #20653: Add `object_type_id` filter for Jobs

pheus opened the pull request:
<!--
    Thank you for your interest in contributing to NetBox! Please note that
    our contribution policy requires that a feature request or bug report be
    approved and assigned prior to opening a pull request. This helps avoid
    waste time and effort on a proposed change that we might not be able to
    accept.

    IF YOUR PULL REQUEST DOES NOT REFERENCE AN ISSUE WHICH HAS BEEN ASSIGNED
    TO YOU, IT WILL BE CLOSED AUTOMATICALLY.

    Please specify your assigned issue number on the line below.
-->
### Fixes: #20653

<!--
    Please include a summary of the proposed changes below.
-->

#### Summary
Fix Jobs list returning empty results when filtering by **Object Type** by introducing a proper `object_type_id` filter and exposing it in the Jobs filter form/fieldsets.

#### Changes
- Add `object_type_id` to the Jobs FilterSet.
- Update the Jobs filter form and fieldsets to include it.
- No model/schema changes or migrations.

arthanson reviewed (changes requested):
Good work, but I think this is still broken in the REST API - would be good to fix that in this PR.  There is an object_type_id in the REST filter but sending the same filter param as in the UI returned 0 results instead of the two that I had in the REST API.

pheus committed:
feat(filtersets): Add `object_type_id` filter for Jobs

Introduce a new `object_type_id` filter to enhance filtering by object
type for Jobs. Update related forms and fieldsets to incorporate the
new filter for better usability and consistency.

Fixes #20653

arthanson reviewed (approved):
Thanks @pheus
